### PR TITLE
Trash post confirm dialog

### LINKED ISSUE #7374: Posts accidentally deleted
### Expected behavior

Not to be able to accidentally delete a post.

### Actual behavior

The user mentioned that holding the phone in the right hand places the "Delete" button in a very precarious location. He accidentally hit the delete button on the post. The issue was then compounded when there was no way to access the trash from the mobile app.

screenshot_20180209-120723|screenshot_20180209-120737

Internal ref: p2-p4a5px-29j

### PR DESCRIPTION
**Pull request by mzorz**
Fixes #7374 

delete_alert

**Notes**
- A slight difference in title from what the issue #7374 describes: this PR uses the Title `Delete post`  and `Delete page` for the dialog, to keep them short and according to what was already being used in the app for other dialogs.

**To test:
CASE A:**
1. go to Posts list
2. try to trash/delete a Post by tapping on the  Trash/delete action icon on the right-bottom corner of the post's card within the list.
3. see the confirmation dialog appear.

**CASE B:**
1. go to Pages list
2. try to trash/delete a Page by tapping on the  Trash/delete action icon on the right-bottom corner of the post's card within the list.
3. see the confirmation dialog appear, and the right wording ("page") is used instead of Post.

### CODE REVIEW

**Comment by mzorz** [comment]
Thank you @mikedang ! 🙇

**Comment by mzorz** [comment]
@akirk would you mind checking if there's something that we might want to change to make i18n easier? thanks in advance 🙇

**Comment by iamthomasbishop** [comment]
A note on this, per a private discussion w/ @diegoreymendez (and the relevant issue: https://github.com/wordpress-mobile/WordPress-iOS/pull/8830):

On iOS, we have filtering on the Posts list to view between Published, Draft, Scheduled and Drafts statuses. This means the Trashed posts are accessible, and therefore deleting a post from the list _is not_ permanent. Therefore, _if we were to also do this on Android_ (and I think we should), we would then have to switch this dialog to say something like:

Delete Post?
This post will be sent to the Trash.
`[Cancel]` `[Delete]`

**Comment by mikedang** [comment]
I'd tweak the copy to:

Delete Post?
This post will be deleted and sent to Trash.

**Comment by iamthomasbishop** [comment]
@mikedang I like that! When we decide to add the filtering to the Posts list (this will probably be in a separate issue), we'll change the copy to this.

**Comment by diegoreymendez** [comment]
Just a note.  Even if the trash filter is not accessible, I'd check to see if posts are effectively deleted permanently.  Considering how that works on a technical leve, there's a chance we're not really deleting posts permanently on Android either.

**Comment by iamthomasbishop** [comment]
@diegoreymendez That's correct, just confirmed on Android - we aren't deleting them permanently on Android either, just sending to the Trash.

**Comment by mzorz** [comment]
Thank you @mikedang @iamthomasbishop @diegoreymendez! changed wording as per discussions in 1bbe08a and 2caed37, reflecting as indicated in https://github.com/wordpress-mobile/WordPress-Android/pull/7392#issuecomment-370928356

Here's how it looks now:

screen shot 2018-03-06 at 11 33 57 pm
screen shot 2018-03-06 at 11 33 42 pm

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -321,13 +321,14 @@
...
-    <string name="delete_post">Delete post</string>
-    <string name="delete_page">Delete page</string>
+    <string name="delete_post">Delete post?</string>
+    <string name="delete_page">Delete page?</string>
     <string name="share_url">Share URL</string>
     <string name="posts_fetching">Fetching posts…</string>
     <string name="pages_fetching">Fetching pages…</string>
     <string name="toast_err_post_uploading">Unable to open post while it\'s uploading</string>
     <string name="dialog_confirm_cancel_post_media_uploading">Are you sure? Deleting this post will also cancel the media upload.</string>
+    <string name="dialog_confirm_delete_post_page">This %s will be deleted and sent to Trash</string>
```
Comment (akirk) [inline WordPress/src/main/res/values/strings.xml]: Please create two strings here:
`This post will be deleted and sent to Trash` and `This page will be deleted and sent to Trash`. "This" might need different translations, depending on the gender of page or post.

Comment (mzorz) [reply WordPress/src/main/res/values/strings.xml]: Thanks for the suggestion, addressed in ec54139

**Comment by 0nko** [comment]
Also looks good. We can :shipit: after the lint is fixed.

**Comment by 0nko** [comment]
There seems to be a conflict.

**Comment by 0nko** [comment]
:shipit:

**Pull request merged**